### ISSUE TITLE
Still alive ?

### ISSUE THREAD

**Issue by Tos26** [seblucas/cops#512]
Hello,

I use COPs for several years. I'm worried about the date of last commit.
Is the project still maintained or the developpers are gone ?

**Comment by marioscube** [comment]
Enjoy it as long as it works. No need to fix that what is not broken.

With a small fix it will work with php8.

I think it will work for a few more years.

**Comment by horus68** [comment]
The Developer has been missing from some time and has not respond some emails of mine.
The software still works and has not a real alternative!
I would like a github fork to be created to implement small fixes available in the comments to make things easier.
I'm not a software developer but I can be involved in the organization of the new project if someone could take care of the software pack release

**Comment by marioscube** [comment]
I think the developer moved on with life. 
He once planned a code overhaul of mainly the bootstrap theme I think. It never was published. Mainly, I think, because the default theme is "just working" (ugly but utilitarian)., so why change a winning team.

If you want the bootstrap theme, then take a look al calibre-web. 

@horus68 
I see only 2 problems with the COPS code:
 - installing from source code gives an error. You need to install an older version of composer to make it work. The latest release still works.
 - there will be an error once we start to use php8 (probably in the next major release of debian 12 in 2023?) This error is easily fixable in the code for everyone who can use a text editor.

Am I missing anything here?

**Comment by horus68** [comment]
Hi, you have a point on the "working app". I would love the bootstrap theme to be implemented and some minor fixes shared by several users to be all implemented in the core pack. As it is now, new users tend to go elsewhere.  But it works!
Note: On the fork: I could even think on putting some money into it in order to make it a clean app ready to install and work from core pack!

**Comment by Tos26** [comment]
Ok, I was afraid of this kind of information :(

> The software still works and has not a real alternative!
Did you test [Calibre-web](https://github.com/janeczku/calibre-web) as an alternative? It seems a well alive project with more features than COPS.

**Comment by marioscube** [comment]
To me COPS has 2 unique selling points:
 - **multiple calibre libraries!**
 - it works on relatively underpowered devices (This week I installed it again on a Raspberry Pi 1b and it worked!) 
COPS was created to be "_a simple tool to be installed on a small server_". 

I like the default theme, bud for those who like the bootstrap theme calibre-web might be an alternative. But as far as I know it misses the multiple libraries and I find it waaaaaay to much (users, options, etc) for my personal needs.

**Comment by horus68** [comment]
> Did you test Calibre-web as an alternative? It seems a well alive project with more features than COPS.

COPS is the way to work as a simple viewer on tiny servers.
For any other features (editing metadata, converting, synch and so on) just use the main Calibre server!

**Comment by Tos26** [comment]
I agree with your comments about the number of features of calibre-web, I don't really need them.
My COPS is currently running on a raspberry PI 3B+, so I will try soon calibre-web on it and I will give you feedback if you are interested.
I didn't know about the multiple libraries on calibre-web, it could be an issue for me...

**Comment by Tos26** [comment]
Just for information:
calibre-web installed on my raspberry PI 3. Nice app, and there is no latency or lag in this config.
I confirm, I don't see multiple libraries management, only shelves.

**Comment by marioscube** [comment]
What is the size of your library?

With just a few books even the slowest Synology 110j is fast enough.

**Comment by Tos26** [comment]
921 books, only epub.

**Comment by marioscube** [comment]
I think, that is close to the sweet spot for COPS and similar apps on a slow device like the raspberry pi.

You can speed up browsing in COPS by caching the thumbnails on disk.

Look for this setting in config_default.php and copy these lines in config_local.php

**Comment by Aspiring-Writer** [comment]
Should I look into Calibre-Web instead of COPS? I want something that I can easily access on my Kobo. I thought COPS was what I was looking for, but if it isn't being maintained and Calibre-Web is better, should I even try COPS?

**Comment by gglanzer** [comment]
What about this program. Would there still be answers there?
https://github.com/rvolz/BicBucStriim

**Comment by marioscube** [comment]
@Aspiring-Writer
I think that COPS (default theme) works very well with Kobo (I have a Kobo Glo HD).

The bootstrap theme of COPS and Calibre-web might be just a bit to much for the Kobo browser.

@gglanzer
BicBuckStream is also not updated since 2020.
I have never really tried it so cannot comment on it.

I would advise to start with COPS and if you really want more options and eyecandy then try calibre-web.

**Comment by horus68** [comment]
Before choosing an alternative to COPS you need to figure out the workflow for your library:
For me its:
- Calibre editor to create a library, manage ebooks, edit metadata, convert, sinch with ereaders (like Kobo using USB)
- To access the Calibre Library: COPS. Note that I use COPS to read a copy of the library folder (books and metadata file from Calibre). My Calibre Library is on a local windows computer, the Cops library is on a web server (the synch goes automatically from Calibre to COPS using a file synchronizer that works via FTP on Windows)

So, to access the Calibre library on a network or online ou need to choose a simple reader like Cops: it works, and you will not use anything else.
You don't need editing features. You will only need the access to read/download books. Everything else needs to be done by Calibre. You shouldn't use any features outside the Calibre software to avoid issues.
But if you want to edit metadata, tags, converting over a network, manage users, then install the proper solution: Calibre Server (installed at the same time you installed Calibre, you just need to activate it and it works perfectly.
Note: many "featured" apps to read Calibre libraries was created before the Calibre Server was availabe and working.

**Comment by marioscube** [comment]
I had a short look at miniCalOPe.
 - could not get it to work in 15 minutes [error 500] (maybe I will need to spend more time with the config files?)
 - basically it is of the same age as COPS. No new options in the last few years.
 - COPS can do OPDS as well. I find integration with calibre important, but YMMV!

**Comment by Aspiring-Writer** [comment]
I couldn't get it to work either :(

At this point, I'm looking into pyShelf and/or coding my own

**Comment by horus68** [comment]
Instead of inventing the wheel with other "not so perfect" solutions, why not just create a "community" fork for COPS to implement the few fixes it requires to use the new Calibre library out of the box? Then it could be directly used by anyone.
Later maybe someone could find the time to update it with an improved template and a PHP8 ready status.

**Comment by marioscube** [comment]
@horus68 
COPS (php7.4) works with calibre 5.37 database out of the box. Or did I miss something?
It only does not work if there is a very. very, old version of sqlite3. 

I only use the default template or OPDS so no real need for "improved templates". If you want a beautiful template then consider calibre-web, but I consider it overkill for my needs.

PHP8 is a real (fixed) issue. If php8 becomes mainstream in Debian then COPS definitely needs an update or a fork.
But maybe the developer will implement that fix himself in COPS if he still uses COPS (probably).

**Comment by Aspiring-Writer** [comment]
@horus68 I'm considering building my own e-book server with Python. I don't know PHP, and it would take me a while to learn, but if you want to do that I'd be happy to test it.

**Comment by therion23** [comment]
Would any of you php-savvy folks be willing to supply a patch that makes it PHP8 compatible? I am a total nincompoop at PHP but i do know how to apply a patch.

**Comment by marioscube** [comment]
See #492

**Comment by therion23** [comment]
Thank you so so so very much, dunno how i missed that but works like a charm!

**Comment by SenorSmartyPants** [comment]
I made a working bootstrap theme

https://github.com/SenorSmartyPants/cops/tree/bootstrap

**Comment by marioscube** [comment]
@SenorSmartyPants 
I will check it out when I set-up mij new ServerPC with COPS next week. :-)

Any more "enhancements" to COPS that you know and think COPS needs??

**Comment by mikespub** [comment]
Included in #522 and release 1.3.2 at https://github.com/mikespub-org/seblucas-cops
